File: scripts/seed.py
import psycopg2
from faker import Faker

# Database connection details
DB_HOST = "localhost"
DB_PORT = "5432"
DB_NAME = "osrd"
DB_USER = "osrd"
DB_PASSWORD = "password"

# Number of entities to create
NUM_ENTITIES = 15

def seed_database():
    try:
        # Connect to the database
        conn = psycopg2.connect(
            host=DB_HOST,
            port=DB_PORT,
            dbname=DB_NAME,
            user=DB_USER,
            password=DB_PASSWORD
        )
        cur = conn.cursor()

        # Initialize Faker
        fake = Faker()

        # Seed authn_user table
        for _ in range(NUM_ENTITIES):
            name = fake.email()
            cur.execute("INSERT INTO authn_user (name) VALUES (%s)", (name,))

        # Seed project table
        for _ in range(NUM_ENTITIES):
            name = fake.sentence(nb_words=3)
            objectives = fake.sentence(nb_words=10)
            description = fake.paragraph(nb_sentences=5)
            creation_date = fake.date_time_this_decade()
            last_modification = fake.date_time_this_decade()
            tags = []
            cur.execute(
                "INSERT INTO project (name, objectives, description, creation_date, last_modification, tags) VALUES (%s, %s, %s, %s, %s, %s)",
                (name, objectives, description, creation_date, last_modification, tags)
            )


        # Commit the changes
        conn.commit()

        print("Database seeded successfully!")

    except (Exception, psycopg2.DatabaseError) as error:
        print(error)
    finally:
        if conn is not None:
            conn.close()

if __name__ == '__main__':
    seed_database()

File: editoast/src/client/seed.rs

use anyhow::Result;
use chrono::{DateTime, Utc};
use database::DbConnectionPoolV2;
use diesel::prelude::*;
use diesel_async::RunQueryDsl;
use fake::faker::internet::en::SafeEmail;
use fake::faker::lorem::en::Sentence;
use fake::Fake;

use database::tables;

#[derive(Insertable)]
#[diesel(table_name = tables::authn_user)]
pub struct NewUser {
    pub name: String,
}

#[derive(Insertable)]
#[diesel(table_name = tables::project)]
pub struct NewProject {
    pub name: String,
    pub objectives: Option<String>,
    pub description: Option<String>,
    pub creation_date: DateTime<Utc>,
    pub last_modification: DateTime<Utc>,
    pub tags: Vec<Option<String>>,
}

pub async fn seed_database(db_pool: DbConnectionPoolV2) -> Result<()> {
    let mut conn = db_pool.get().await?;

    // Seed users
    for _ in 0..15 {
        let name: String = SafeEmail().fake();
        let new_user = NewUser { name };
        diesel::insert_into(tables::authn_user::table)
            .values(&new_user)
            .execute(&mut conn)
            .await?;
    }

    // Seed projects
    for _ in 0..15 {
        let name: String = Sentence(1..3).fake();
        let new_project = NewProject {
            name,
            objectives: Some(Sentence(5..10).fake()),
            description: Some(Sentence(10..20).fake()),
            creation_date: Utc::now(),
            last_modification: Utc::now(),
            tags: Vec::<Option<String>>::new(),
        };
        diesel::insert_into(tables::project::table)
            .values(&new_project)
            .execute(&mut conn)
            .await?;
    }

    println!("Database seeded successfully!");

    Ok(())
}
